motion/thruster_allocator/src/allocator_ros.cpp

```diff
@@ -0,0 +1,94 @@
+#include "thruster_allocator/allocator_ros.hpp"
+#include "thruster_allocator/allocator_utils.hpp"
+#include "thruster_allocator/pseudoinverse_allocator.hpp"
+#include "vortex_msgs/msg/thruster_forces.hpp"
+
+Allocator::Allocator() : Node("thrust_allocator_node") {
+  subscription_ = this->create_subscription<geometry_msgs::msg::Wrench>(
+      "thrust/wrench_input", 1,
+      std::bind(&Allocator::wrench_callback, this, std::placeholders::_1));
+
+  publisher_ = this->create_publisher<vortex_msgs::msg::ThrusterForces>(
+      "thrust/thruster_forces", 1);
+
+  Eigen::MatrixXd thrust_configuration_pseudoinverse;
+  if (!calculatePseudoinverse(thrust_configuration,
+                              &thrust_configuration_pseudoinverse)) {
+    RCLCPP_FATAL(get_logger(), "Failed to compute pseudoinverse of thrust "
+                               "config matrix. Killing node...");
+    rclcpp::shutdown();
+  }
+
+  // Set T_pinv in m_pseudoinverse_allocator
+  m_pseudoinverse_allocator.reset(
+      new PseudoinverseAllocator(thrust_configuration_pseudoinverse));
```

Comment:
`new`: Dynamically allocating memory without freeing it is very very bad. In this case only one variable is allocated once, which is fine. But in general it is best to just use `std::shared_ptr`